python/py-coolprop/Portfile

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+github.setup        CoolProp CoolProp 6f4dfa82c1232f4827a81040284262a7099ed3fe
+revision            1
+fetch.type          git
+
+checksums           rmd160  e09935be3be632239bf631e5bea3e54fd68f39e1 \
+                    sha256  fd26827cdb70ffcef657538a2e1105d6a46beda6d898c385b96961e5b46e15e8 \
+                    size    8002925
+
+
+name                py-coolprop
+categories-append   science
+license             MIT
+maintainers         openmaintainer @jsalort
+platforms           darwin
+
+description         Open-source thermodynamic and transport properties database
+long_description    CoolProp implements: Pure and pseudo-pure fluid equations \
+                    of state and transport properties for 122 components, \
+                    mixture properties using high-accuracy Helmholtz energy \
+                    formulations, correlations of properties of incompressible \
+                    fluids and brines, computationally efficient tabular \
+                    interpolation, highest accuracy psychrometric routines, \
+                    user-friendly interface around the full capabilities of NIST \
+                    REFPROP, fast IAPWS-IF97 (Industrial Formulation) for \
+                    Water/Steam, and Cubic equations of state (SRK, PR).
+homepage            http://www.coolprop.org
+
+python.versions     38 39 310
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+        path:bin/cmake:cmake \
+        port:py${python.version}-setuptools \
+        port:py${python.version}-cython
+    
+    depends_test-append \
```

Comment:
are these really only test dependencies and none of them are runtime dependencies? 

If they are indeed only needed for tests then you should just remove them since you do not define a test-phase (also then delete the commented out test-phase stuff)

Reply:
I am not sure what is the policy in this case. They are dependencies useful at runtime naturally. But they are not hard dependencies. They are needed only if one uses the GUI functions of CoolProp (which I don't personnaly). It happens that the test suite also tests the GUI, therefore they are required for the test suite to pass.

However, there are a number of issues with the test suites, as well as the GUI. I am under the impression that the GUI and the test suite are not well maintained, and I am not convinced of their usefulness in the present state: they have errors due to outdated Python 2.7 code. I started to have a look (hence the patches) and I thought of submitting them upstream eventually, but I am not sure if it is really worth the work.

I thought I would let them commented out, so that it is easier if someone eventually decides to try again fixing the test suite. But if it is not the policy to leave half-working things commented out, then I can certainly remove them.